optimizations-server/src/sft/common.py

```diff
@@ -0,0 +1,64 @@
+import random
+import typing as t
+from typing import Any, List, Tuple, TypedDict
+
+from pydantic import BaseModel, Field
+
+
+class Model(BaseModel):
+    displayName: str
+    name: str
+    provider: t.Literal["openai", "fireworks"]
+
+
+class FineTuningRequest(BaseModel):
+    kind: t.Literal["sft"]
+    function: str
+    # TODO - require nonempty
```

Comment:
why? we take none in the UI

Reply:
The node sever requires it to be either `null` or a non-empty string. I don't think it's a huge deal either way, since an empty string will just result in an error message later on.